modules/articles/tests/server/admin.article.server.routes.tests.js

```diff
@@ -0,0 +1,278 @@
+    });
+
+    // Save a user to the test db and create new article
+    user.save(function () {
+      article = {
+        title: 'Article Title',
+        content: 'Article Content'
+      };
+
+      done();
+    });
+  });
+
+  it('should be able to save an article if logged in', function (done) {
+    agent.post('/api/auth/signin')
+      .send(credentials)
+      .expect(200)
+      .end(function (signinErr, signinRes) {
+        // Handle signin error
+        if (signinErr) {
+          return done(signinErr);
+        }
+
+        // Get the userId
+        var userId = user.id;
+
+        // Save a new article
+        agent.post('/api/articles')
+          .send(article)
+          .expect(200)
+          .end(function (articleSaveErr, articleSaveRes) {
+            // Handle article save error
+            if (articleSaveErr) {
+              return done(articleSaveErr);
+            }
+
+            // Get a list of articles
+            agent.get('/api/articles')
+              .end(function (articlesGetErr, articlesGetRes) {
+                // Handle article save error
+                if (articlesGetErr) {
+                  return done(articlesGetErr);
+                }
+
+                // Get articles list
+                var articles = articlesGetRes.body;
+
+                // Set assertions
+                (articles[0].user._id).should.equal(userId);
+                (articles[0].title).should.match('Article Title');
+
+                // Call the assertion callback
+                done();
+              });
+          });
+      });
+  });
+
+  it('should be able to update an article if signed in', function (done) {
+    agent.post('/api/auth/signin')
+      .send(credentials)
+      .expect(200)
+      .end(function (signinErr, signinRes) {
+        // Handle signin error
+        if (signinErr) {
+          return done(signinErr);
+        }
+
+        // Get the userId
+        var userId = user.id;
+      
+        // Save a new article
+        agent.post('/api/articles')
+          .send(article)
+          .expect(200)
+          .end(function (articleSaveErr, articleSaveRes) {
+            // Handle article save error
+            if (articleSaveErr) {
+              return done(articleSaveErr);
+            }
+
+            // Update article title
+            article.title = 'WHY YOU GOTTA BE SO MEAN?';
+
+            // Update an existing article
+            agent.put('/api/articles/' + articleSaveRes.body._id)
+              .send(article)
+              .expect(200)
+              .end(function (articleUpdateErr, articleUpdateRes) {
+                // Handle article update error
+                if (articleUpdateErr) {
+                  return done(articleUpdateErr);
+                }
+
+                // Set assertions
+                (articleUpdateRes.body._id).should.equal(articleSaveRes.body._id);
+                (articleUpdateRes.body.title).should.match('WHY YOU GOTTA BE SO MEAN?');
```

Comment:
I merely ported some of these tests over from the Articles tests. So we were probably already using `match` here. I could always change it.